app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapFormViewModel.kt

```
@@ -594,11 +597,42 @@ internal class SwapFormViewModel @Inject constructor(
     private suspend fun checkTokenSelectionResponse(targetArg: String) {
-        val result = requestResultRepository.request<Coin>(targetArg)?.id
+        val result = requestResultRepository.request<AssetSelected>(targetArg) ?: return
+
+        if (result.isDisabled) {
+            uiState.update { it.copy(isLoading = true) }
+            vaultId?.let {
+                try {
+                    val account = accountsRepository.loadAccount(vaultId!!, result.token)
+                    updateAccountInAddresses(account)
+                    uiState.update { it.copy(isLoading = false) }
+                } catch (t: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)